crate/kmip/src/crypto/wrap/unwrap_key.rs

```diff
@@ -106,6 +107,11 @@ pub(crate) fn unwrap(
         ciphertext.len()
     );
 
+    // Make sure that the key used to unwrap can be used to unwrap.
+    unwrapping_key
+        .attributes()?
+        .is_usage_mask_flag_set(CryptographicUsageMask::UnwrapKey)?;
+
```

Comment:
What's the downside with current state of the code ?
It looks like two acceptable ways of achieving the same thing or do I miss something ?

Reply:
Your code is really fine. I hesitated to post this comment. But since you use this bit quite a few times, a dedicated function could be defined:
- pros: you do not need to understand what these three lines of code are doing, each time you see them; using a properly named function would make it obvious (you felt the need to make it obvious via a comment). You save three lines per occurrence.
- cons: you define yet another function (this should not be a real cons: having many small functions is *good*).
Whether to change it or not is up to you. This is merely a suggestion.

Reply:
It is a dedicated function plus it is a method for the `Attributes` struct.
I agree on the naming tho.